audb/core/api.py

```diff
@@ -278,8 +285,17 @@ def dependencies(
                     version,
                     verbose=verbose,
                 )
-                deps.load(os.path.join(tmp_root, define.DEPENDENCIES_FILE))
-                deps.save(deps_path)
+                # Load parquet or csv from tmp dir
+                # and store as pickle in cache
+                deps_path = os.path.join(tmp_root, define.DEPENDENCIES_FILE)
+                legacy_path = os.path.join(tmp_root, define.LEGACY_DEPENDENCIES_FILE)
+                cached_path = os.path.join(db_root, define.CACHED_DEPENDENCIES_FILE)
+                if os.path.exists(deps_path):
+                    deps.load(deps_path)
+                else:
+                    deps.load(legacy_path)
+                # Store as pickle in cache
+                deps.save(cached_path)
```

Comment:
I assmume that this is where the cached fle gets written as pkl (instead of parquet).

Performance: the benchmark that you report says that loading performance from cache is worse for parquet than for  pkl, effectively motivating to contingue to persist cache as pickle. However, is the increase between pickle (0.0016/0.0021s. versus 0.0048 s when caching with parquet significant, and does it outweigh the advantages of parquet over pickle? I am not thinking so much about the standard flaws of pickle, rather than interoperatibility with other programming languages (that will be able to read parquet but not pickle).

Reply:
I did a proper benchmarking with more datasets and 10 runs each. I added the results to the description and list them here as well:

| dataset | before | pull request | parquet |
| --- | --- | --- | --- |
| mozillacommonvoice | 0.465±0.027 s | 0.257±0.033 s | 0.613±0.025 s |
| voxceleb2 | 0.234±0.008 s | 0.127±0.008 s | 0.280±0.011 s |
| librispeech | 0.071±0.007 s | 0.034±0.005 s | 0.086±0.007 s |
| imda-nsc-read-speech-balanced | 0.494±0.018 s | 0.283±0.036 s | 0.552±0.016 s |
| emodb | 0.008±0.018 s | 0.006±0.011 s | 0.010±0.012 s |

Using pickle files in cache is faster as using parquet. Using parquet would even be slower than our current solution, which is not acceptable in my opinion.

Reply:
Cool. The current PR uses pickle format now - which is the fastest. As no backward compatibility is required, this is fine for the dependencies. So this can be resolved.